### DISCUSSION TITLE
How do I export the Node json to Flask backend?

### DISCUSSION THREAD

**Discussion by ethanpng2021** [xyflow/xyflow#1065] [Q&A]
The DragNDrop codes have this:

```
const onDrop = (event: DragEvent) => {
    event.preventDefault();
    
    if (reactFlowInstance) {
      const type = event.dataTransfer.getData('application/reactflow');
      const position = reactFlowInstance.project({ x: event.clientX, y: event.clientY - 40 });
      **const newNode: Node = {
        id: getId(),
        type,
        position,
        data: { label: `${type} node` },
      };**
      
      setElements((es) => es.concat(newNode));
    }
  };

  return (

    <div className="dndflow">
      <ReactFlowProvider>
        <div className="reactflow-wrapper">
          <ReactFlow
            elements={elements}
            onConnect={onConnect}
            onElementsRemove={onElementsRemove}
            onLoad={onLoad}
            onDrop={onDrop}
            onDragOver={onDragOver}
          >
            <Controls />
          </ReactFlow>
        </div>
        <Sidebar />
      </ReactFlowProvider>
    </div>
  );
};

export default DnDFlow;
```

However, I want to export the Node like this to Flask:

https://jasonwatmore.com/post/2020/02/01/react-fetch-http-post-request-examples

I can't find any sample codes that deal with this. That's why I came here to ask.

Warm regards,
Ethan

**Comment by moklick** [comment]
Hey @ethanpng2021 Sorry but I don't get your question. What do you want to achieve? What do you mean by "export to Flask"?

**Reply by ethanpng2021** [reply]
Hi Moklick, I need to generate a json file from the DragNDrop and pass to Flask.

By exporting to json, I mean this:
E.g. https://gojs.net/latest/samples/flowchart.html

Your DragNDrop json format data does show up on the Javascript console when I press F12, but I need to 'POST' or pass them to Flask. Flask will receive the json using "jsondata = request.get_json()"

Yours gratefully,
Ethan

**Reply by leejustin** [reply]
@ethanpng2021 If you're asking about how to get the data as an JSON payload, you can use `toObject()` or `getElements()` (https://reactflow.dev/docs/api/react-flow-instance/).

**Comment by ethanpng2021** [comment]
Hi Justin,

Thanks a lot. You saved me!

Warm regards,
Ethan
________________________________
